docs/day-2-operations/Backups.md

```diff
@@ -0,0 +1,272 @@
+# 3scale Backup & Restore
+
+## Overview
+
+This section intends to provide an Operator of a 3scale installation the information they need to:
+
+* Setup the backup procedures for persistent data
+* Perform a restore from backup of the persistent data
+
+Such that, in the case of a failure, they can restore 3scale to correctly running operational state and continue operating.
+
+### Persistent Volumes
+In a 3scale deployment on OpenShift all persistent data is stored either in a storage service in the cluster (not currently used), a Persistent Volume (PV) provided to the cluster by the underlying infrastructure or a storage service external to the cluster (be that in the same Data Center or elsewhere).
+
+### Considerations
+The backup and restore procedures for persistent data are required to vary depending on the storage used, to ensure the backups and restores preserve data conistsency (e.g. that a partial write, or a partial transaction is not captured). i.e. It is not sufficient to backup the underlying Persistent Volumes for a database, but instead the databases backup mechanisms should be used.
+
+Also, some parts of the data are synchronized between different components. One copy is considered the "source of truth" for the data set, and the other a copy that is not modified locally, just synchronized from the "source of truth". In these cases, upon restore, the "source of truth" should be restored and then the copies in other components synchronized from it.
+
+## Data Sets
+This section goes into more detail on the different data sets in the different persistent stores, their purposes, the storage type used and if it is the "source of truth" or not.
+
+The full state of a 3scale deployment is stored across these services and their persistent volumes:
+
+* `system-mysql`:  Mysql database (`mysql-storage`)
```

Comment:
Should we consider other supported database too? i.e. Oracle for now

Reply:
@hallelujah I'm basing myself on the productized provided templates: 

- https://github.com/3scale/3scale-amp-openshift-templates/tree/master/amp

As Oracle is proprietary software, I guess that the Oracle should provide support for backup and restore of their database over Openshift. 

But I guess we need to find some way to communicate that.

Reply:
@andrewdavidmackenzie thoughts about that?

Reply:
Only if something particular about the DB changes the overall strategy. Then (for me) would be acceptable to do "do a similar procedure for the other Supported DBs, using the documentation from the DB provider" and *maybe* link to their detailed docs. Avoid reproducing the DB docs here, but help the user find them.

Reply:
Ok, I will try to look for Oracle Containerized DB backup documentation. (Not sure If it is public though.)

Reply:
Guys I was reviewing how we support Oracle in 3scale, and what we do is ask for an Oracle database endpoint and create the system Image with the required Oracle Client files. We don't provide a template or a DeployConfig for Oracle system.

This means that we do not control or provide a recommendation for an environment where the Oracle Database will be deployed. So I think it does make sense to provide backup documentation for this, as the deployment environment could range from  AIX to Windows...